var/spack/repos/builtin/packages/py-bluepymm/package.py

```diff
@@ -0,0 +1,46 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyBluepymm(PythonPackage):
+    """Blue Brain Model Management Python Library"""
+
+    homepage = "https://github.com/BlueBrain/BluePyMM"
+    url = "https://pypi.io/packages/source/b/bluepymm/bluepymm-0.6.35.tar.gz"
+
+    version('0.6.35', '8455e4543057e2a4889cce46bfe841a4041eaa039d6aece1ac2ccfdb755c1ccf')
+    
+    depends_on('py-setuptools', type='build')
+    depends_on('py-bluepyopt', type='run')
+    depends_on('py-matplotlib', type='run')
+    # THe below dependency should disappear once the matplotlib package is fixed
+    depends_on('py-backports-functools-lru-cache', type='run')
+    depends_on('py-pandas', type='run')
+    depends_on('py-numpy', type='run')
+    depends_on('py-ipyparallel', type='run')
+    depends_on('py-lxml', type='run')
+    depends_on('py-sh', type='run')
+    depends_on('neuron', type='run')
```

Comment:
I wonder if you want to add `mpi` as variant if non-mpi neuron can be used (of course if its useful for users)

Reply:
bluepymm never needs mpi-enabled neuron, so it doesn't seem to useful to add that variant for now

Reply:
@wvangeit : in that case, should we change : 

```
depends_on('neuron', type='run')
```

to

```
depends_on('neuron~mpi', type='run')
```

this is because `mpi` variant is `True` by default in neuron package :

```
py-bluepyopt@1.6.56%clang@9.0.0-apple+neuron arch=darwin-sierra-x86_64
    ^neuron@develop%clang@9.0.0-apple+binary+coreneuron~cross-compile~debug+mpi+multisend~profile+pysetup+python~rx3d+shared arch=darwin-sierra-x86_64
        ^autoconf@system%clang@9.0.0-apple arch=darwin-sierra-x86_64
        ^automake@system%clang@9.0.0-apple arch=darwin-sierra-x86_64
        ^bison@system%clang@9.0.0-apple arch=darwin-sierra-x86_64
        ^flex@system%clang@9.0.0-apple+lex arch=darwin-sierra-x86_64
        ^libtool@system%clang@9.0.0-apple arch=darwin-sierra-x86_64
        ^mpich@3.2%clang@9.0.0-apple device=ch3 +hydra netmod=tcp patches=e591891fae6a1d029b95edbc93865c57d165784e77cf8b5822ae7d014b137a03 +pmi+romio~verbs arch=darwin-sierra-x86_64
```
this will avoid building mpi package (and reduce build time).

Reply:
Would this mean bluepymm won't use neuron+mpi when it is installed ?

I would actually rather suggest to disable the mpi in the neuron package by default. Wouldn't it in general be better to follow the defaults in the 'configure' of the original package ?

Reply:
> Would this mean bluepymm won't use neuron+mpi when it is installed ? 

yes

> I would actually rather suggest to disable the mpi in the neuron package by default. > Wouldn't it in general be better to follow the defaults in the 'configure' of the original package ?

As spack packages are "commonly" in use in the cluster environment, `+mpi` is default across many packages. To be consistent, I would keep this default for now. We can revisit this in near future.

Reply:
its ok to keep as it is....typically we use pre-installed mpi and hence it doesn't make difference.